daal4py/sklearn/model_selection/split.py

```diff
@@ -0,0 +1,201 @@
+                    dtypes[i] = 0
+                elif dtype in [np.float32, 'float32']:
+                    dtypes[i] = 1
+                elif dtype in [np.float64, 'float64']:
+                    dtypes[i] = 2
+                else:
+                    fallback = True
+                    break
+        if fallback:
+            train = train.reshape((n_train,))
+            test = test.reshape((n_test,))
+            res.append(_safe_indexing(arr, train))
+            res.append(_safe_indexing(arr, test))
+        else:
+            train = train.reshape(n_train, 1)
+            test = test.reshape(n_test, 1)
+
+            if len(arr.shape) == 2:
+                n_cols = arr.shape[1]
+                reshape_later = False
+            else:
+                n_cols = 1
+                reshape_later = True
+
+            unique_dtypes = set(dtypes)
+            dtypes_list = dtypes
+            if origin_type == 'pandas DataFrame':
+                if len(unique_dtypes) > 1:
+                    origin_type += ' with multiple dtypes'
+                else:
+                    origin_type += ' with one dtype'
+            if len(dtypes) == 1:
+                dtypes = dtypes * n_cols
+            dtypes = np.array(dtypes, dtype=np.int32, ndmin=2)
+
+            if origin_type == 'numpy':
+                order = 'C' if arr.flags['C_CONTIGUOUS'] == True else 'F'
+
+                train_arr = np.empty(shape=(n_train, n_cols), dtype=arr.dtype, order=order)
+                test_arr = np.empty(shape=(n_test, n_cols), dtype=arr.dtype, order=order)
+
+                d4p.daal_train_test_split(arr.reshape((arr.shape[0], n_cols), order='A'), train_arr, test_arr, train, test, dtypes)
+
+                if reshape_later:
+                    train_arr = train_arr.reshape((train_arr.shape[0],))
+                    test_arr = test_arr.reshape((test_arr.shape[0],))
+                    arr = arr.reshape((arr.shape[0],), order='A')
+
+            elif origin_type == 'pandas DataFrame with one dtype':
+                order = 'C' if arr.values.flags['C_CONTIGUOUS'] == True else 'F'
+
+                train_arr = np.empty(shape=(n_train, n_cols), dtype=arr.values.dtype, order=order)
+                test_arr = np.empty(shape=(n_test, n_cols), dtype=arr.values.dtype, order=order)
+
+                d4p.daal_train_test_split(arr.values, train_arr, test_arr, train, test, dtypes)
+
+                train_arr, test_arr = pd.DataFrame(train_arr), pd.DataFrame(test_arr)
+
+            elif origin_type == 'pandas DataFrame with multiple dtypes':
+                train_arr, test_arr = {}, {}
+                for i, col in enumerate(list(arr.columns)):
+                    train_arr[col] = np.empty(shape=(n_train,), dtype=arr.dtypes[col])
+                    test_arr[col] = np.empty(shape=(n_test,), dtype=arr.dtypes[col])
+                train_arr = pd.DataFrame(train_arr)
+                test_arr = pd.DataFrame(test_arr)

```

Comment:
Does Sklearn have similar logic?

Reply:
sklearn uses specific indexing function which is based mostly on iloc method of pd.DataFrame, for example.